accepted/00000-maintenance-windows.md

```diff
@@ -0,0 +1,409 @@
+
+If a Maintenance Schedule is assigned and which one, should be visible in the systems details page.
+Implementation of XMLRPC functions for getting and setting the current maintenance schedule must be implemented.
+For setting a schedule should be possible for multiple systems using an array of system ids.
+
+## System in Maintenance Mode
+
+The current status of the system should be reflected in the User Interface and API.
+
+If a system is in Maintenance Mode, the User Interface could show a special icon [ || ] (pause symbol of an audio player).
+
+In the XMLRPC API a flag could be added to `system.getDetails()`.
+A special function `system.listSystemsInMaintenanceMode()` should be added as well.
+
+
+## Recurring Actions
+
+Recurring Actions are scheduled using a cron like syntax. It may not be aligned to a
+Maintenance Schedule defined via ICalendar.
+
+When it happens that an Recurring Action is executed outside of a maintenance window, it should be skipped.
+A warning should be logged and maybe a Notification raised.
+
+The customer need to take care to align the two schedules.
+
+Define recurring Maintenance Windows and align them with Recurring Actions.
+When a Maintenance Window in such a series was excluded and an alternative window added, a one time highstate action must be scheduled by the admin.
+
+Randomly appearing Maintenance Windows cannot be handled.
+
+
+### Traditional Clients
+
+We will not implement something special for traditional clients. For the first iteration we take care
+that actions are scheduled only inside of maintenance windows. This would apply also for traditional clients.
+
+As long as the traditional client uses osad, or `rhn_check` is executed inside of the maintenance window
+the feature would work for them as well.
+
+We will document in the manual, that for traditional clients actions are also executed outside of Maintenance Windows
+if `rhn_check` is executed only at a later point in time.
+
+## What 3rd party tools to create schedules
+
+We require customers to use 3rd party tools to create the ICalendar files.
+We need to test the most common used tools to generate them.
+
+- office 365 / outlook
+- google calendar
+- confluence ?
+
+For Uyuni users Linux tools might be interesting
+- korganizer
+- Lightning (Thunderbird Extension)
+- evolution
+
+
+## Notifications
+
+When a system is assigned to a maintenance schedule, every finished action should check, if the system is still in maintenance mode.
+If this is not the case it means, the action was started inside of the maintenance window, but was not able to finish in time.
+This Maintenance Window overflow should be show with a notification.
+Ideally we should have only one notification per maintenance window, not per client which cause the overflow.
+It is not an `error`, but it show the admin that the window might needs to get increased for the next runs.
+
+
+## Second Iteration
+
+### Implement a Maintenance Mode
```

Comment:
"Maintenance mode" has been mentioned multiple times before, and now there is "Implement a Maintenance Mode" section. I think it'd be better to explicitly mention, that this is the maintenance mode "override" (or "flag").

```suggestion
### Implement a Maintenance Mode override
```